src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -344,45 +389,36 @@ private async Task<bool> ValidateCredentialsWork(Uri remoteUri, ICredential cred
             // TODO: ideally we'd also check if the credentials have expired based on some local metadata
             // (once/if we get such metadata storage), and return false if they have.
             // This would be more efficient than having to make REST API calls to check.
-
+            Uri remoteUri = input.GetRemoteUri();
             _context.Trace.WriteLineSecrets($"Validate credentials ({credentials.Account}/{{0}}) are fresh for {remoteUri} ...", new object[] { credentials.Password });
 
-            if (!IsBitbucketOrg(remoteUri))
-            {
-                // TODO: Validate DC/Server credentials before returning them to Git
-                // Currently credentials for DC/Server are not checked by GCM.
-                // Instead the validation relies on Git to try and fail with the credentials and then request GCM to erase them
-                _context.Trace.WriteLine("For DC/Server skip validating existing credentials");
-                return await Task.FromResult(true);
-            }
-
             // Bitbucket supports both OAuth + Basic Auth unless there is explicit GCM configuration.
             // The credentials could be for either scheme therefore need to potentially test both possibilities.
             if (SupportsOAuth(authModes))
             {
                 try
                 {
-                    await ResolveOAuthUserNameAsync(credentials.Password);
+                    await ResolveOAuthUserNameAsync(input, credentials.Password);
                     _context.Trace.WriteLine("Validated existing credentials using OAuth");
                     return true;
                 }
-                catch (Exception)
+                catch (Exception ex)
                 {
-                    _context.Trace.WriteLine("Failed to validate existing credentials using OAuth");
+                    _context.Trace.WriteLine($"Failed to validate existing credentials using OAuth: {ex.Message}");
```

Comment:
Note that we also have a utility method on `ITrace` for tracing exceptions.

```suggestion
                    _context.Trace.WriteLine($"Failed to validate existing credentials using OAuth");
                                  _context.Trace.WriteException(ex);
```